core/lib/l1_contract_interface/src/i_executor/structures/commit_batch_info.rs

```diff
@@ -101,6 +103,72 @@ impl<'a> Tokenizable for CommitBatchInfo<'a> {
                 ),
                 // `systemLogs`
                 Token::Bytes(serialize_commitments(&self.0.header.system_logs)),
+            ]
+        }
+    }
+
+    pub fn into_tokens_calldata(&self) -> Token {
+        let mut tokens = self.as_tokens_base();
+        let mut pubdata = self
+            .0
+            .header
+            .pubdata_input
+            .clone()
+            .unwrap_or(self.0.construct_pubdata());
+
+        let kzg_settings = load_kzg_settings();
+        let mut blob = pubdata.clone();
+        blob.resize(ZK_SYNC_BYTES_PER_BLOB, 0u8);
+        let blob_commitment = KzgInfo::new(&kzg_settings, blob).to_blob_commitment();
+
+        pubdata.extend(blob_commitment);
+
+        pubdata.insert(0, 0);
```

Comment:
there are at least two `.clone()`s here and one `insert` that will likely have to move the whole thing in memory. I would suggest to at least make `KzgInfo::new` accept the blob by reference since it only uses it by reference it looks like.